### ISSUE TITLE
`getTransaction` throws when a transaction is not yet confirmed

### ISSUE THREAD

**Issue by keefertaylor** [ripple/ripple-lib#1094]
To reproduce:
1) submit a transaction, receive a hash
2) call `getTransaction` with the given hash, *before* transaction is confirmed

Expected:
- Get all fields of the transaction besides ones that can only be populated when the transaction is included in a ledger. 

Actual: 
- Throws an error: 
```
(node:70113) UnhandledPromiseRejectionWarning: MissingLedgerHistoryError: Server is missing ledger history in the specified range
```

At the very least, this should throw a better error. I'd expect this to adhere to the semantics of the `tx` RPC though, which would be:
- returning an appropriate `engineResultCode`
- returning a `validated` boolean

**Comment by intelliot** [comment]
@keefertaylor Good points. At the time this method was created (which predates my time working on this library) the decision was made to only return **validated** ledger data. This has the potential to reduce confusion for users: if `getTransaction` succeeds, then the transaction has been validated.

For reliable transaction submission, it's important to know which ledgers are searched when you query for a transaction. That way, you can know if the transaction was not, and never will be, included in a validated ledger -- which is one of the failure modes for a transaction, and must be detected/handled.

That's why, in typical usage, you should provide the [`minLedgerVersion` and `maxLedgerVersion` parameters](https://xrpl.org/rippleapi-reference.html#parameters-9) when you call `getTransaction`. These values should indicate the range of ledger versions that your transaction could possibly have been included in. In other words, set `minLedgerVersion` to what the validated ledger version was **before** you submitted your transaction, and `maxLedgerVersion` to your transaction's `LastLedgerSequence`. With these parameters provided appropriately, `getTransaction` will tell you, definitively, if your transaction failed to be included in a validated ledger. And of course, if the transaction is found in a validated ledger, then `getTransaction` will give you the transaction's final status.

In other words, `getTransaction` was never intended to be a mere passthrough for the `tx` RPC.

Rather, it tries to make the user's life easier by internally tracking and considering the ledger history that the rippled server has.

If you want to use the `tx` RPC, you should do that with [request](https://xrpl.org/rippleapi-reference.html#request). Something like:
```js
const tx = await api.request('tx', { /* ... */ })
```

Does that make sense?

**Comment by keefertaylor** [comment]
Apologies for lack of response. `api.request` is what I needed.
